src/scheduler/scheduler.directive.js

```diff
@@ -0,0 +1,101 @@
+(function(){
+    'use strict';
+
+    angular.module('motech-scheduler')
+        .directive('collapser', function($compile) {
+        return {
+            restrict: 'C',
+            scope: {
+                collapser: '='
+            },
+            link: function (scope, element, attrs) {
+                element.bind('click', function() {
+                    var target = $(scope.collapser),
+                        shown = target.hasClass('in');
+
+                    if (shown) {
+                        $(scope.collapser).collapse('toggle');
+                        element.find('.fa-caret-down').removeClass("fa-caret-down").addClass("fa-caret-right");
+                        target.find('.collapse').collapse('hide');
+                        target.find('.fa-caret-down').removeClass("fa-caret-down").addClass("fa-caret-right");
+                    } else {
+                        element.find('.fa-caret-right').removeClass("fa-caret-right").addClass("fa-caret-down");
+                        $(scope.collapser).collapse('toggle');
+                    }
+                });
+            }
+        };
+    });
+
+    angular.module('motech-scheduler')
+        .directive("mtInvalid", function() {
+        return {
```

Comment:
This one is used in create-or-update-job page. This one turns labels (next to fields which are necessary for a specific job type) red when those fields are empty. This is a pattern from the Motech repo. Should I remove it then?